### PR TITLE
feat(PLAY-324): upgrade @flowplayer/player to latest canary, fix opts prop types

### PR DESCRIPTION
**Pull request by maximzavadskiy**
- upgrade @flowplayer/player to 3.10.4-rc.4
  - fix opts prop types
  - add typespec tests

### CODE REVIEW

File: demo/demo.tsx [hunk 1]
```diff
@@ -61,7 +61,7 @@ const Main = () => {
             opts={{
               title: "Example title",
               description: "Example description",
-              poster: { },
+              poster: "https://picsum.photos/200/300",
```
Comment (rashadatjou) [inline demo/demo.tsx:64]: 😂

**Review by rashadatjou — COMMENTED** [review]
Great that you added a test for this!

File: src/flowplayer.tsx [hunk 2]
```diff
@@ -4,11 +4,14 @@ import type { Config, ConfigWith } from "@flowplayer/player";
...
 import flowplayer from "@flowplayer/player";
 
+type KeyValue = Record<string, any>;
+
 // - Types
 type Props = {
   token: Config["token"],
   src: Config["src"],
-  opts?: Omit<ConfigWith<any>, "token" | "src">
+  // allow custom plugin config
```
Comment (sdvassil) [inline src/flowplayer.tsx:13]: @maximzavadskiy Is this something we should document?

**Comment by maximzavadskiy** [comment]
@rashadatjou good to merge?

**Comment by rashadatjou** [comment]
@maximzavadskiy let me test this out locally and if everything is functioning I will give you the green light

**Review by rashadatjou — APPROVED** [review]
@maximzavadskiy the changes look great I just cleaned up the `test/index.test-d.tsx` file. This PR is good to be merged.

**Pull request merged**